src/avian/classpath-common.h

```diff
@@ -103,6 +103,9 @@ void arrayCopy(Thread* t,
           } else {
             throwNew(t, GcIndexOutOfBoundsException::Type);
           }
+        } else if (LIKELY(length < 0)) {
+          throwNew(t, GcNegativeArraySizeException::Type, "%d", length);
```

Comment:
According to https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#arraycopy(java.lang.Object,%20int,%20java.lang.Object,%20int,%20int), this should throw an `IndexOutOfBoundsException` if the length is negative.

Reply:
Good catch! I've made the change to adhere to the Oracle docs.